conan/tools/microsoft/nmaketoolchain.py

```diff
@@ -15,42 +16,122 @@ def __init__(self, conanfile):
         :param conanfile: ``< ConanFile object >`` The current recipe object. Always use ``self``.
         """
         self._conanfile = conanfile
-        self._environment = None
 
-    @property
-    def cl_flags(self):
-        cppflags = []
+        # Flags
+        self.extra_cflags = []
+        self.extra_cxxflags = []
+        self.extra_ldflags = []
+        self.extra_defines = []
+
+        # Defines
+        self._ndebug = None
         build_type = self._conanfile.settings.get_safe("build_type")
-        if build_type in ['Release', 'RelWithDebInfo', 'MinSizeRel']:
-            cppflags.append("/DNDEBUG")
+        if build_type in ["Release", "RelWithDebInfo", "MinSizeRel"]:
+            self._ndebug = "NDEBUG"
+
+        self._build_type_flags = build_type_flags(self._conanfile.settings)
+        self._build_type_link_flags = build_type_link_flags(self._conanfile.settings)
 
-        bt = build_type_flags(self._conanfile.settings)
-        if bt:
-            cppflags.extend(bt)
+        self._cppstd = cppstd_flag(self._conanfile.settings)
+        self._msvc_runtime_flag = self._get_msvc_runtime_flag()
 
-        cppstd = cppstd_flag(self._conanfile.settings)
-        if cppstd:
-            cppflags.append(cppstd)
-        from conan.tools.microsoft import msvc_runtime_flag
+    def _get_msvc_runtime_flag(self):
```

Comment:
Seems unnecessary, the formatting of flag is doing nothing, the ``msvc_runtime_flag()`` can be used directly instead of this method.

Reply:
It does something, it prepends `/`. It's just a copy paste of an autotoolstoolchain private method.